src/Library/demos/Wallpaper/code.rs

```diff
@@ -0,0 +1,36 @@
+use std::error::Error;
+
+use crate::workbench;
+use ashpd::desktop::wallpaper::SetOn;
+use ashpd::desktop::wallpaper::WallpaperRequest;
+use glib::MainContext;
+use gtk::glib;
+use gtk::prelude::*;
+
+pub fn main() {
+    let button: gtk::Button = workbench::builder().object("button").unwrap();
+    button.connect_clicked(|_| {
+        MainContext::default().spawn_local(async {
+            if let Err(err) = set_wallpaper().await {
+                eprintln!("Could not set wallpaper: {err}")
+            }
+        });
+    });
+}
+
+async fn set_wallpaper() -> Result<(), Box<dyn Error>> {
+    let uri = url::Url::parse(&workbench::resolve("./wallpaper.png"))?;
+    let request = WallpaperRequest::default()
+        .set_on(SetOn::Both)
+        .show_preview(true)
+        .build_uri(&uri)
+        .await?;
+
+    if request.response().is_ok() {
```

Comment:
this is not great as you are now missing the response related errors :) so just call resquest.response()?;